postmaster/models.py

```diff
@@ -366,6 +371,17 @@ def set_password(self, new_password):
 
         self.password = bcrypt.generate_password_hash(new_password)
 
+    def generate_otp_secret(self, **kwargs):
+        # generate a random secret
+        self.otp_secret = base64.b32encode(urandom(10)).decode('utf-8')
+
+    def get_totp_uri(self):
```

Comment:
No reason to throw an exception, since the user will never be calling this function without generate_otp_secret being ran first. The exception would be for the developer, in which case we wouldn't want to catch it.